assemble-and-restart

```diff
@@ -6,10 +6,10 @@ set -eo pipefail
 # source files, we use that instead.
 if [ -f /opt/app-root/src/.s2i/bin/assemble ]; then
     /opt/app-root/src/.s2i/bin/assemble
-elif [ -f /usr/local/s2i/assemble ]; then # java s2i
+elif [ -z "${ODO_S2I_SCRIPTS_URL}" ]; then # For S2I scripts path, use the env var set by odo if not available in component source
```

Comment:
This doesn't seem to work for me. 
I am using the openjdk18 builder and image and while `ODO_S2I_SCRIPTS_URL` is correctly set to `/usr/local/s2i`, this bash test doesn't seem to pass.

When I do `odo push` I get:

```
Pushing changes to component: backend
Please wait, building component....
+ set -eo pipefail
+ '[' -f /opt/app-root/src/.s2i/bin/assemble ']'
+ '[' -z /usr/local/s2i ']'
+ /usr/libexec/s2i/assemble
/var/lib/supervisord/bin/assemble-and-restart: line 14: /usr/libexec/s2i/assemble: No such file or directory
failed to push component: backend
```

Reply:
Basically this check will only pass if the env var `ODO_S2I_SCRIPTS_URL` does not exist :P

Reply:
So basically I think it should be:

```
elif [ -z "${ODO_S2I_SCRIPTS_URL}" ]; then 
    /usr/libexec/s2i/assemble
else
    rm -rf /opt/app-root/src/.git # ensure we don't copy git files since they can cause problems
    cp -a /opt/app-root/src/. /tmp/src # java s2i expects the source to be under /tmp/src
    ${ODO_S2I_SCRIPTS_URL}/assemble
fi
```

Reply:
Tested the above change and it worked for me

Reply:
aah weird I tested this(but without the only comment I added later in this PR) for a wildfly component but that worked not sure if it hit the first case there

I'll reverify the PR thanks @geoand

Reply:
:+1:

Reply:
@geoand I think its now fixed can you re-try it once now

Reply:
Cool, I tried it and it worked for me using the java s2i image